crates/ra_hir_expand/src/db.rs

```diff
@@ -72,6 +72,8 @@ pub trait AstDatabase: SourceDatabase {
 
     #[salsa::interned]
     fn intern_macro(&self, macro_call: MacroCallLoc) -> LazyMacroId;
+    fn macro_arg_text(&self, id: MacroCallId) -> Option<GreenNode>;
```

Comment:
Should this be an `Arc`? It uses 1 MB less memory with `Arc<GreenNode>`.

Reply:
`GreenNode` has an internal Arc-like thing. Not sure why wrapping it in another `Arc` would save memory.